feat(A529yyds):add pgsql operation functions:register, login and chan…

A529yyds opened the pull request:
…gerpassword

fix(A529yyds):fixed pgsqlClient __conn initialization

A529yyds committed:
feat(A529yyds):add pgsql operation functions:register, login and changerpassword
fix(A529yyds):fixed pgsqlClient __conn initialization

yushou-cell commented on src/Order.cpp:
```diff
@@ -110,12 +139,144 @@ nlohmann::json Redis::SetCheckCode::constructResponse(const nlohmann::json &orde
+            result["correct"] = 1;
+        }
+    }
+    LOG(ERROR) << result.dump();
```
There is a problem here, both success and failure will print error logs

yushou-cell commented on src/Order.cpp:
```diff
@@ -110,12 +139,144 @@ nlohmann::json Redis::SetCheckCode::constructResponse(const nlohmann::json &orde
+    }
+    catch(const pqxx::sql_error &e)
+    {
+        LOG(ERROR) << "SQL error: " << e.what() << std::endl;
```
In this exception, a JSON command-reply needs to be constructed indicating execution failure, with success=0 and correct=0. The same applies below.

yushou-cell commented on src/Order.cpp:
```diff
@@ -110,12 +139,144 @@ nlohmann::json Redis::SetCheckCode::constructResponse(const nlohmann::json &orde
 
+bool SQL::SQLOperation::isOrderRight(const nlohmann::json &order, std::string orType)
+{
+    return order.find("order").value().get<std::string>() == orType;
```
The function ConstructResponseMsgPgSQL has been validated. There’s no need for validation here. This function can be removed

A529yyds replied:
OK

A529yyds committed:
fix{A529yyds):fix some redundant code
-Remove debug code: LOG(ERROR) << result.dump();
-Remove duplicate function and its application code

yushou-cell reviewed (approved):
Approve
